HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/activities/ExecuteActivity.kt

```diff
@@ -334,15 +308,99 @@ class ExecuteActivity : BaseActivity(), Entrypoint {
+        } else {
+            showWifiDialogIfNeeded()
+        }
+
+    private fun showWifiDialogIfNeeded(): Completable = Completable.create { emitter ->
+        if (getCurrentSsid() == shortcut.ssid) {
+            finishActivityIfNeeded()
+            emitter.onComplete()
+        } else {
+            showWifiPickerConfirmation(emitter)
+        }
+    }
+
+    private fun getCurrentSsid(): String {
+        val wifiManager = getSystemService(WIFI_SERVICE) as WifiManager
+        val info = wifiManager.connectionInfo
+        Log.d("Wifi Connection", "Ssid: ${info.ssid}")
+        return info.ssid.removePrefix("\"").removeSuffix("\"")
+    }
+
+    private fun showWifiPickerConfirmation(emitter: CompletableEmitter) {
+        DialogBuilder(context)
+            .title(shortcutName)
+            .message(String.format(getString(R.string.message_wrong_wifi_network), shortcut.ssid))
+            .dismissListener {
+                emitter.cancel()
+            }
+            .positive(getString(R.string.action_label_select)) {
+                showWifiPicker()
+                emitter.cancel()
+            }
+            .negative(R.string.dialog_cancel)
+            .showIfPossible()
+            ?: run {
+                emitter.cancel()
+            }
+    }
+
+
+    private fun showWifiPicker() {
+        val intent = Intent(WifiManager.ACTION_PICK_WIFI_NETWORK)
+        intent.putExtra("extra_prefs_show_button_bar", true)
+        intent.putExtra("wifi_enable_next_on_connect", true)
+
+        startActivity(intent)
+    }
+
+    private fun requestPermissionsForWifiCheckIfNeeded(): Completable =
+        if (shortcut.ssid.isEmpty())
+            Completable.complete()
+        else
+            showRequestPermissionRationalIfNeeded().concatWith(requestPermissionsForWifiCheck())
+
+
+    private fun finishActivityIfNeeded() {
+        if (shouldFinishImmediately()) {
+            finishWithoutAnimation()
+        }
+        showProgress()
+    }
+
+    private fun showRequestPermissionRationalIfNeeded(): Completable =
+       if(shouldShowRequestPermissionRationale(Manifest.permission.ACCESS_FINE_LOCATION))
+           DialogBuilder(context)
+               .title(getString(R.string.title_permission_dialog))
+               .message(getString(R.string.message_permission_rational))
+               .positive(R.string.dialog_ok)
+               .showAsCompletable()
+       else
+           Completable.complete()
+
+    private fun requestPermissionsForWifiCheck() = Completable.create { emitter ->
+        RxPermissions(this)
+            .request(Manifest.permission.ACCESS_FINE_LOCATION)
+            .subscribe { granted ->
+                if (granted) {
+                    Log.d("Wifi Connection", "Ppermission ACCESS_FINE_LOCATION granted")
+                    emitter.onComplete()
+                } else {
+                    emitter.onError(MissingPermissionException({ "Could not get location permission. Aborting request." }))
```

Comment:
The error message should not be hardcoded but localized instead, see my other comment for `MissingPermissionException`.